File: source/core/sr/connection_event_args.cpp
//
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
//
// connection_event_args.h: Implementation declarations for CSpxConnectionEventArgs C++ class
//

#include "stdafx.h"
#include "connection_event_args.h"


namespace Microsoft {
namespace CognitiveServices {
namespace Speech {
namespace Impl {


CSpxConnectionEventArgs::CSpxConnectionEventArgs()
{
}

const std::wstring& CSpxConnectionEventArgs::GetSessionId()
{
    SPX_IFTRUE_THROW_HR(m_sessionId.length() == 0, SPXERR_UNINITIALIZED);
    return m_sessionId;
}

void CSpxConnectionEventArgs::Init(const std::wstring& sessionId)
{
    SPX_IFTRUE_THROW_HR(m_sessionId.length() != 0, SPXERR_ALREADY_INITIALIZED);
    m_sessionId = sessionId;
}

} } } } // Microsoft::CognitiveServices::Speech::Impl

File: source/bindings/java/com/microsoft/cognitiveservices/speech/ConnectionEventArgs.java
//
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
//
package com.microsoft.cognitiveservices.speech;

import com.microsoft.cognitiveservices.speech.util.Contracts;

/**
 * Defines payload for connection events like Connected/Disconnected.
 * Added in version 1.2.0
 */
public final class ConnectionEventArgs extends SessionEventArgs
{
    /*! \cond INTERNAL */

    /**
     * Constructs an instance of a ConnectionEventArgs object.
     * @param eventArgs internal connection event handle.
     */
    public ConnectionEventArgs(long eventArgs) {
        super(eventArgs);
        storeEventData(false);
    }

    ConnectionEventArgs(long eventArgs, boolean dispose) {
        super(eventArgs);
        storeEventData(dispose);
    }

    private void storeEventData(boolean disposeNativeResources) {
        if (disposeNativeResources == true) {
            super.close();
        }
    }

    /*! \endcond */
}
